src/hyperlight_host/src/sandbox/sandbox_builder.rs

```diff
@@ -0,0 +1,1308 @@
+    pub flags: MemoryRegionFlags,
+    /// Offset of the memory section in the guest's address space
+    pub page_aligned_guest_offset: usize,
+    /// Host address of the memory section
+    pub host_address: Option<usize>,
+    /// Size of the memory section
+    pub page_aligned_size: usize,
+}
+
+/// Holds the memory sections of the sandbox.
+/// We use a BTreeMap to keep the sections sorted by offset.
+#[derive(Clone)]
+pub(crate) struct SandboxMemorySections {
+    /// A map of memory sections, sorted by offset
+    pub sections: BTreeMap<usize, SandboxMemorySection>,
+}
+
+impl Debug for SandboxMemorySections {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_map().entries(self.sections.iter()).finish()
+    }
+}
+
+impl SandboxMemorySections {
+    pub(crate) fn new() -> Self {
+        Self {
+            sections: BTreeMap::new(),
+        }
+    }
+
+    pub(crate) fn get_guest_code_offset(&self) -> Option<usize> {
+        self.sections
+            .iter()
+            .find(|(_, section)| section.name == DEFAULT_GUEST_CODE_SECTION_NAME)
+            .map(|(_, section)| section.page_aligned_guest_offset)
+    }
+
+    pub(crate) fn get_hyperlight_peb_section_offset(&self) -> Option<usize> {
+        self.sections
+            .iter()
+            .find(|(_, section)| section.name == DEFAULT_HYPERLIGHT_PEB_SECTION_NAME)
+            .map(|(_, section)| section.page_aligned_guest_offset)
+    }
+
+    pub(crate) fn get_hyperlight_peb_section_host_address(&self) -> Option<usize> {
+        self.sections
+            .iter()
+            .find(|(_, section)| section.name == DEFAULT_HYPERLIGHT_PEB_SECTION_NAME)
+            .map(|(_, section)| {
+                #[allow(clippy::unwrap_used)]
+                // this is safe because we set the host addresses when creating the uninitialized sandbox
+                section.host_address.unwrap()
+            })
+    }
+
+    pub(crate) fn get_tmp_stack_section_offset(&self) -> Option<usize> {
+        self.sections
+            .iter()
+            .find(|(_, section)| section.name == DEFAULT_TMP_STACK_SECTION_NAME)
+            .map(|(_, section)| section.page_aligned_guest_offset)
+    }
+
+    pub(crate) fn get_paging_structures_offset(&self) -> Option<usize> {
+        self.sections
+            .iter()
+            .find(|(_, section)| section.name == DEFAULT_PAGING_STRUCTURES_SECTION_NAME)
+            .map(|(_, section)| section.page_aligned_guest_offset)
+    }
+
+    pub(crate) fn get_custom_guest_memory_section_offset(&self) -> usize {
+        let offset = self
+            .sections
+            .iter()
+            .find(|(_, section)| section.name == DEFAULT_CUSTOM_GUEST_MEMORY_SECTION_NAME)
+            .map(|(_, section)| section.page_aligned_guest_offset);
+
+        #[allow(clippy::unwrap_used)]
+        // this is safe because create the custom guest memory section w/ the SandboxBuilder usage
+        offset.unwrap()
```

Comment:
why are we using unwrap in here? Applies to all uses. Is there are reason why we would want to crash the host in these cases?